### ISSUE TITLE
Network Module Extension

### ISSUE THREAD

**Issue by noahcoetsee** [WhitestormJS/whitestorm.js#396]
Hey guys!

Just here letting you know that I'm continuing the original development of the Server design for WHS. I'm specifically building the framework off of a pre-existing technical framework. Photon Networking Utilities has been in the market for a while, and the techniques they use for internal/external networking is extremely polished (even more so than the default Unity Networking), therefore inclining me to base the feature-set off of specific designs from Photon.

 Anyways, I want to restart the Alpha-Network-Module branch (with the exception of taking some notes about creating modules and such), so after confirmation, I'd like @sasha240100 to please help delete and restore that branch as a fork of the dev build.

I hope this was clear, and you guys are ready to start the new year with some great work on this great library.

Looking forward to 2019,
Noah

**Comment by hirako2000** [comment]
> Photon Networking Utilities has been in the market for a while

Do you mean [this](https://www.photonengine.com/)? 
I came across [Lance](https://github.com/lance-gg/Lance), worth looking at it too as it runs on node.js with socket.io

> I want to restart the Alpha-Network-Module branch

Just create a new branch, did Alex give you rights to create/delete branches in the whs repo?

**Comment by noahcoetsee** [comment]
Yes, he did, but I guess I need to save some parts of my previous work. I'll be doing that.

**Comment by noahcoetsee** [comment]
Instead of proceeding with a branch, I went ahead and created a new module repository for this (it seems necessary since this feature is obviously outside the scope of WebGL rendering and such).

https://github.com/WhitestormJS/network-module

I need some help converting it to a real module. I've never used babel before, or yarn or any other dev tools involved besides NodeJS and NPM. So, if one of you would help out, could you please setup the project for module development? I need some help getting all the tools and stuff ready.

@sasha240100 @hirako2000 

I really appreciate any help xD. Thanks, guys!

**Comment by sasha240100** [comment]
@FeaturedSpace DM us in discord for any help. We're always welcome!;)

**Comment by hirako2000** [comment]
Yes! The API looks great as currently shown on the branch readme. Let's talk on Discord to complete this effort!
